src/neo-vm/ExecutionEngine.cs

```diff
@@ -696,22 +696,42 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                         break;
                     case OpCode.PACK:
                         {
-                            int size = (int)context.EvaluationStack.Pop().GetBigInteger();
-                            if (size < 0 || size > context.EvaluationStack.Count)
+                            StackItem item = context.EvaluationStack.Pop();
+                            if (item is VMArray array)
                             {
-                                State |= VMState.FAULT;
-                                return;
+                                List<byte> nba = new List<byte>();
+                                for (int i = 0; i < array.Count; i++)
+                                    nba.AddRange(array[i].GetByteArray());
+                                byte[] ba = nba.ToArray();
+                                context.EvaluationStack.Push(ba);
+                            }
+                            else
+                            {
+                                int size = (int)item.GetBigInteger();
+                                if (size < 0 || size > context.EvaluationStack.Count)
+                                {
+                                    State |= VMState.FAULT;
+                                    return;
+                                }
+                                List<StackItem> items = new List<StackItem>(size);
+                                for (int i = 0; i < size; i++)
+                                    items.Add(context.EvaluationStack.Pop());
+                                context.EvaluationStack.Push(items);
                             }
-                            List<StackItem> items = new List<StackItem>(size);
-                            for (int i = 0; i < size; i++)
-                                items.Add(context.EvaluationStack.Pop());
-                            context.EvaluationStack.Push(items);
                         }
                         break;
                     case OpCode.UNPACK:
                         {
                             StackItem item = context.EvaluationStack.Pop();
-                            if (item is VMArray array)
+                            if (item is ByteArray iba)
+                            {
+                                byte[] ba = iba.GetByteArray();
+                                List<StackItem> items = new List<StackItem>(ba.Length);
+                                for (int i = 0; i < ba.Length; i++)
+                                    items.Add(new ByteArray(new byte[]{ba[i]}));
```

Comment:
A byte array should be unpacked to BigInteger?

Reply:
I guess it should be unpacked to several bytearrays... then if pack again, the same bytearray is obtained. If unpack to biginteger, some byte values will change (e.g. 0xff to 0xff00)